test/IntegrationTests/Helpers/MockMetricsCollector.cs

```diff
@@ -0,0 +1,174 @@
+// <copyright file="MockMetricsCollector.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using Google.Protobuf;
+using IntegrationTests.Helpers.Models;
+using Opentelemetry.Proto.Collector.Metrics.V1;
+using Xunit.Abstractions;
+
+namespace IntegrationTests.Helpers;
+
+public class MockMetricsCollector : IDisposable
```

Comment:
Right now we use 2 separate collectors for testing traces and metrics. When we start testing traces using OTLP do we think it will be better to configure 2 independent collectors on different ports, or do we think we should use a single listener on the same port? I think that decision affects whether or not we should rename this file. Another alternative, could be using a single OTLP collector but have the code for traces and metrics in separate partial classes.

Reply:
I am leaning towards having 2 independent collectors on different ports. I think it will make tests readable. Also, we won't end up with a big "god" collector. It will make also the listeners independent so e.g. a bug in the trace handler will not affect the metrics handler.

Reply:
One of the current downsides to having 2 independent collectors is that the OTLP exporter doesn't support telemetry specific environment variable configuration yet. So if we need to support sending the telemetry to 2 different OTLP endpoints we need to implement that support ourselves and then get the SDK to adopt those telemetry specific variables. Since this is something that we know we want to do anyways, it's not a big deal. It's just slightly more scope being added to one of the next 2 releases.

With the refactoring in this PR you already structured the code in such a way that it is easier to share code between the different listeners.

Reply:
I think that whatever way we go regarding OTLP mock collector, that this refactoring is still beneficial. It can be used e.g. to create a MockJaegerCollector.